node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java

```diff
@@ -49,7 +50,7 @@ public static String readString(ByteBuffer buffer) {
   public static List<String> readStringList(ByteBuffer buffer) {
     int size = readInt(buffer);
     if (size < 0) {
-      return null;
+      return Collections.emptyList();
```

Comment:
You need to check for upper-level dependencies on these nulls, such as changing == null to isEmpty when serializing

Reply:
Already removed